Sources/Signal.swift

```diff
@@ -2015,177 +1876,352 @@ private enum ThrottleWhileState<Value> {
 	}
 }
 
+private protocol SignalAggregateStrategy {
+	/// Update the latest value of the signal at `position` to be `value`.
+	///
+	/// - parameters:
+	///   - value: The latest value emitted by the signal at `position`.
+	///   - position: The position of the signal.
+	///
+	/// - returns: `true` if the aggregating signal should terminate as a result of the
+	///            update. `false` otherwise.
+	mutating func update(_ value: Any, at position: Int) -> Bool
+
+	/// Record the completion of the signal at `position`.
+	///
+	/// - parameters:
+	///   - position: The position of the signal.
+	///
+	/// - returns: `true` if the aggregating signal should terminate as a result of the
+	///            completion. `false` otherwise.
+	mutating func complete(at position: Int) -> Bool
+
+	init(count: Int, action: @escaping (ContiguousArray<Any>) -> Void)
+}
+
 extension Signal {
+	private struct CombineLatestStrategy: SignalAggregateStrategy {
+		private struct Placeholder {}
+
+		private var values: ContiguousArray<Any>
+		private var completionCount: Int
+		private var _isInitialized: Bool
+		private let action: (ContiguousArray<Any>) -> Void
+
+		private var isInitialized: Bool {
+			mutating get {
+				if _isInitialized {
+					return true
+				}
+
+				_isInitialized = values.reduce(true, { $0 && !($1 is Placeholder) })
+				return _isInitialized
+			}
+		}
+
+		mutating func update(_ value: Any, at position: Int) -> Bool {
+			values[position] = value
+
+			if isInitialized {
+				action(values)
+			}
+
+			return false
+		}
+
+		mutating func complete(at position: Int) -> Bool {
+			completionCount += 1
+			return completionCount == values.count
+		}
+
+		init(count: Int, action: @escaping (ContiguousArray<Any>) -> Void) {
+			values = ContiguousArray(repeating: Placeholder(), count: count)
+			completionCount = 0
+			_isInitialized = false
+			self.action = action
+		}
+	}
+
+	private struct ZipStrategy: SignalAggregateStrategy {
+		private var values: ContiguousArray<[Any]>
+		private var isCompleted: ContiguousArray<Bool>
+		private let action: (ContiguousArray<Any>) -> Void
+
+		mutating func update(_ value: Any, at position: Int) -> Bool {
+			values[position].append(value)
+
+			if values.reduce(true, { $0 && !$1.isEmpty }) {
+				var buffer = ContiguousArray<Any>()
+				buffer.reserveCapacity(values.count)
+
+				for index in values.startIndex ..< values.endIndex {
```

Comment:
`values.indices` holds a strong reference to the collection, and is not recommended if you gonna do in-place mutation.